File: app/src/main/java/linusfessler/alarmtiles/shared/TimeOfDayFormatter.java
package linusfessler.alarmtiles.shared;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;
import java.util.Locale;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class TimeOfDayFormatter {

    @Inject
    public TimeOfDayFormatter() {

    }

    public String format(final int hourOfDay, final int minutesOfHour, final boolean is24Hours) {
        final Calendar calendar = new GregorianCalendar(0, 0, 0, hourOfDay, minutesOfHour);
        final Date timeOfDay = calendar.getTime();
        final SimpleDateFormat displayFormat = getFormat(is24Hours);
        return displayFormat.format(timeOfDay);
    }

    private SimpleDateFormat getFormat(final boolean is24Hours) {
        if (is24Hours) {
            return new SimpleDateFormat("H:mm", Locale.UK);
        }

        return new SimpleDateFormat("h:mm a", Locale.US);
    }
}

File: app/src/main/java/linusfessler/alarmtiles/shared/TimeOfDayFormatter.kt
package linusfessler.alarmtiles.shared

import java.text.SimpleDateFormat
import java.util.*
import javax.inject.Inject
import javax.inject.Named
import javax.inject.Singleton

@Singleton
class TimeOfDayFormatter @Inject constructor(@Named("is24Hours") is24Hours: Boolean) {
    private val format = if (is24Hours) {
        SimpleDateFormat("H:mm", Locale.UK)
    } else {
        SimpleDateFormat("h:mm a", Locale.US)
    }

    fun format(hourOfDay: Int, minutesOfHour: Int): String {
        val calendar = GregorianCalendar(0, 0, 0, hourOfDay, minutesOfHour)
        val timeOfDay = calendar.time
        return format.format(timeOfDay)
    }
}